datafusion/functions-aggregate/src/count.rs

```diff
@@ -148,6 +155,15 @@ impl AggregateUDFImpl for Count {
         "count"
     }
 
+    // In AggregateFunctionPlanner, wildcard is converted to count(1)
+    //
+    // count() -> count(1)
```

Comment:
We still can't run `select count(), count(*)`.
```sh
> select count(), count(*);
Error during planning: Projections require unique expression names but the expression "count(*)" at position 0 and "count(*)" at position 1 have the same name. Consider aliasing ("AS") one of them
```
I suspect that using aliases to restore the original names is a simpler fix. I tried doing this on https://github.com/jonahgao/datafusion/commit/08206fd5179da6efbc12ec2f863d4b3bf5baa6ef.

Reply:
I think this is quite different than the test covered in extended test.

`select count() * count(*);` is executable now.

But I agree, this query should be executable too

Reply:
BTW I verified that both of those queries run in `datafusion` 44 and 45 but does not run on main. Thus this is a regression.

I agree with @jayzhan211  that the issue is different than what is causing the sqlite tests to fail in main

I have filed a ticket to track this:
- https://github.com/apache/datafusion/issues/14855

Reply:
I think they have the same root cause, which is the rewriting by `AggregateFunctionPlanner` introducing duplicate names, and they could all be fixed by using aliases. The old `CountWildcardRule` used `NamePreserver` to achieve a similar effect.